src/ui/public/notify/notifier.js

```diff
@@ -63,7 +63,7 @@ function timerCanceler(notif, cb = _.noop, key) {
 function startNotifTimer(notif, cb) {
   const interval = 1000;
 
-  if (notif.lifetime === Infinity || notif.lifetime === 0) {
+  if (notif.lifetime === 'Infinity' || notif.lifetime === Infinity || notif.lifetime === 0) {
```

Comment:
This doesn't seem like the right place to handle this case.  The reason the string 'Infinity' is intended to be supported in this PR is because of a use case with the advanced settings UI itself, so it seems like this value should be serialized there long before it gets passed to the notifier.